software/winrm/src/main/java/org/apache/brooklyn/location/winrm/WinRmMachineLocation.java

```diff
@@ -80,6 +81,9 @@
     public static final ConfigKey<String> OPERATION_TIMEOUT = newConfigKeyWithPrefix(BrooklynConfigKeys.BROOKLYN_WINRM_CONFIG_KEY_PREFIX, WinRmTool.OPERATION_TIMEOUT);
     public static final ConfigKey<Integer> RETRIES_OF_NETWORK_FAILURES = newConfigKeyWithPrefix(BrooklynConfigKeys.BROOKLYN_WINRM_CONFIG_KEY_PREFIX, WinRmTool.RETRIES_OF_NETWORK_FAILURES);
     public static final ConfigKey<Boolean> USE_HTTPS_WINRM = WinRmTool.USE_HTTPS_WINRM;
+    public static final ConfigKey<String> VERIFY_WINDOWS_UP = newStringConfigKey("verifyWindowsUp",
+            "Sometimes restart could happen after a Windows VM is provisioned. This could be because of System Upgrade or other." +
+            " Brooklyn will wait 5 minutes for a restart to happen before returning the machine as accessible.", null);
```

Comment:
Improve description. Remove mention of 5 minutes - you are not setting a default value.
Also the code could block for double the time set in this config if the machine doesn't come back.